python/beeai_framework/runnable.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+from abc import ABC, abstractmethod
+from collections.abc import Awaitable, Callable
+from typing import Any, TypedDict, Unpack
+
+from pydantic import BaseModel
+from typing_extensions import ParamSpec, TypeVar
+
+from beeai_framework.backend import AnyMessage
+from beeai_framework.context import Run, RunContext, RunMiddlewareType
+from beeai_framework.emitter import Emitter
+from beeai_framework.utils import AbortSignal
+from beeai_framework.utils.dicts import exclude_keys
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+class RunnableOptions(TypedDict, total=False):
+    """Options for a runnable."""
+
+    signal: AbortSignal
+    """The runnable's abort signal data"""
+
+    context: dict[str, Any]
+    """Context can be used to pass additional context to runnable"""
+
+
+class RunnableOutput(BaseModel):
+    """Runnable output."""
+
+    output: list[AnyMessage]
+    """The runnable output"""
+
+    context: dict[str, Any] | None = None
+    """Context can be used to return additional data by runnable"""
+
+
+class Runnable(ABC):
+    """A unit of work that can be invoked using a stable interface.
+
+    For convenience, concrete subclasses can decorate `run` with `runnable_entry`
+    to automatically wrap the runnable into an execution context:
+
+        @runnable_entry
+        async def run(self, input: str | list[AnyMessage], /, **kwargs: Unpack[AgentOptions]) -> RunnableOutput:
```

Comment:
AgentOptions is not defined, I suggest to completely remove this comment and mention this in the documentation

Reply:
Updated